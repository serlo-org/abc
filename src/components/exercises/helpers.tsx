import { addIndex, forEach, map, uniq } from 'ramda';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import AbstractExercise from '../../../packages/entities/exercises/AbstractExercise';
import { Fixtures } from '../../../packages/entities/exercises/helpers';
import loadSound from '../../assets/sounds';
// @ts-ignore: TODO: migrate to TypeScript
import { play } from '../../helpers/audio';
// @ts-ignore: TODO: migrate to TypeScript
import { LoadSounds } from '../helpers/Audio';
import Container from '../screens/Exercise';

const mapIndexed = addIndex(map);

const wrapExercise = <Props, State>(
  Exercise: new (props: Props) => AbstractExercise<Props, State>,
  Component: React.ComponentType<{
    showFeedback: boolean;
    state: State;
    setState: (state: State) => void;
  }>,
  props: Props,
  state?: State
  /* tslint:disable-next-line:no-any TODO: */
): ((props: any) => React.ReactElement<any>) => {
  const exercise = new Exercise(props);

  /* tslint:disable-next-line:no-any TODO: */
  return (additionalProps: any) => (
    <Container
      {...additionalProps}
      submitted={state}
      exercise={exercise}
      Component={Component}
    />
  );
};

const getPropsFromFixtures = <Props, State>(
  fixtures: Fixtures<Props, State>
): Props[] => uniq(map(({ props }) => props, fixtures));

interface IArgs<Props, State> {
  Exercise: new (props: Props) => AbstractExercise<Props, State>;
  Component: React.ComponentType<{
    showFeedback: boolean;
    state: State;
    setState: (state: State) => void;
  }>;
  fixtures: Fixtures<Props, State>;
}

export const createElementsFromFixtures = <Props, State>({
  Exercise,
  Component,
  fixtures
}: IArgs<Props, State>): Array<{
  name: string;
  /* tslint:disable-next-line:no-any TODO: */
  createElement: (props: any) => React.ReactElement<any>;
}> => [
  ...mapIndexed((props, index) => {
    const createElement = wrapExercise(Exercise, Component, props);

    return { name: `default ${index}`, createElement };
  }, getPropsFromFixtures(fixtures)),
  ...map(({ name, state, props }) => {
    const createElement = wrapExercise(Exercise, Component, props, state);

    return { name, createElement };
  }, fixtures)
];

export const createStoriesFromFixtures = <Props, State>(
  /* tslint:disable-next-line:no-any TODO: */
  story: any,
  args: IArgs<Props, State>
): void => {
  forEach(({ name, createElement }) => {
    story.add(name, () => (
      <LoadSounds
        sounds={[loadSound.correct(), loadSound.wrong()]}
        /* tslint:disable-next-line:no-any TODO: */
        render={([correctSound, wrongSound]: any[], soundsLoaded: boolean) =>
          soundsLoaded &&
          createElement({
            onCorrect: () => {
              play(correctSound);
            },
            onWrong: () => {
              play(wrongSound);
            }
          })
        }
      />
    ));
  }, createElementsFromFixtures(args));
};

const doNothing = () => {
  // do nothing
};

export const createTestsFromFixtures = <Props, State>(
  args: IArgs<Props, State>
): void => {
  describe('renders without crashing', () => {
    forEach(({ name, createElement }) => {
      describe(name, () => {
        const tree = renderer.create(
          createElement({
            onCorrect: doNothing,
            onWrong: doNothing
          })
        );
        expect(tree.toJSON()).toMatchSnapshot();
      });
    }, createElementsFromFixtures(args));
  });
};
