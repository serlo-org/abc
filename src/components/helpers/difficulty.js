import React from 'react';

export const createWithDifficulty = generateDifficulty => C => {
  const difficulty = generateDifficulty();

  const WithDifficultyComponent = props => (
    <C {...props} difficulty={difficulty} />
  );

  return WithDifficultyComponent;
};

// TODO: fancy logic
const difficultyGenerator = () => 1/3

export default createWithDifficulty(difficultyGenerator)
