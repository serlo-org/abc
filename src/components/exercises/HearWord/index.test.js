import * as HearWord from '.';

import { createTestsFromFixtures } from '../helpers';

// Jest needs at least one test
it('', () => {
  // empty
});

createTestsFromFixtures(HearWord);
