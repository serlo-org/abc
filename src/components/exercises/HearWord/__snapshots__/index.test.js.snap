// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00b4d5",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "height": "50%",
            "justifyContent": "center",
          },
          Object {
            "height": "35%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
          }
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.1)",
              "borderRadius": 9999999,
              "margin": 20,
            }
          }
        >
          <Image
            collapsable={undefined}
            resizeMode="cover"
            source={1337}
            style={
              Object {
                "borderRadius": 100,
                "height": 200,
                "margin": 20,
                "width": 200,
              }
            }
          />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginLeft": -70,
              "marginRight": 20,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#73dbff",
                "borderRadius": 9999,
                "elevation": 10,
                "height": 55,
                "justifyContent": "center",
                "padding": 5,
                "shadowColor": "rgba(0, 0, 0, 0.1)",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 4,
                },
                "shadowOpacity": 1,
                "shadowRadius": 0,
                "width": 55,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": "50%",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderColor": "#ffffff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Object {
                  "borderRadius": 7,
                  "margin": 5,
                  "padding": 5,
                },
              ],
              Object {},
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": NaN,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": NaN,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              Apfel
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderColor": "#ffffff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Object {
                  "borderRadius": 7,
                  "margin": 5,
                  "padding": 5,
                },
              ],
              Object {},
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": NaN,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": NaN,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              Nase
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderColor": "#ffffff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Object {
                  "borderRadius": 7,
                  "margin": 5,
                  "padding": 5,
                },
              ],
              Object {},
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": NaN,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": NaN,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              Ball
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "top": 35,
        },
        Object {
          "right": 15,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#73dbff",
            "borderRadius": 9999,
            "elevation": 10,
            "height": 40,
            "justifyContent": "center",
            "padding": 5,
            "shadowColor": "rgba(0, 0, 0, 0.1)",
            "shadowOffset": Object {
              "height": 4,
              "width": 4,
            },
            "shadowOpacity": 1,
            "shadowRadius": 0,
            "width": 40,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[` 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#b5ce4d",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "height": "50%",
            "justifyContent": "center",
          },
          Object {
            "height": "35%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
          }
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.1)",
              "borderRadius": 9999999,
              "margin": 20,
            }
          }
        >
          <Image
            collapsable={undefined}
            resizeMode="cover"
            source={1337}
            style={
              Object {
                "borderRadius": 100,
                "height": 200,
                "margin": 20,
                "width": 200,
              }
            }
          />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginLeft": -70,
              "marginRight": 20,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#73dbff",
                "borderRadius": 9999,
                "elevation": 10,
                "height": 55,
                "justifyContent": "center",
                "padding": 5,
                "shadowColor": "rgba(0, 0, 0, 0.1)",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 4,
                },
                "shadowOpacity": 1,
                "shadowRadius": 0,
                "width": 55,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": "50%",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderColor": "#ffffff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Object {
                  "borderRadius": 7,
                  "margin": 5,
                  "padding": 5,
                },
              ],
              Object {},
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": NaN,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": NaN,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              Apfel
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderColor": "#73dbff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Object {
                  "borderRadius": 7,
                  "margin": 5,
                  "padding": 5,
                },
              ],
              Object {
                "backgroundColor": "#ffffff",
              },
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": NaN,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": NaN,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#73dbff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              Nase
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderColor": "#ffffff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Object {
                  "borderRadius": 7,
                  "margin": 5,
                  "padding": 5,
                },
              ],
              Object {},
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": NaN,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": NaN,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              Ball
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "top": 35,
        },
        Object {
          "right": 15,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#73dbff",
            "borderRadius": 9999,
            "elevation": 10,
            "height": 40,
            "justifyContent": "center",
            "padding": 5,
            "shadowColor": "rgba(0, 0, 0, 0.1)",
            "shadowOffset": Object {
              "height": 4,
              "width": 4,
            },
            "shadowOpacity": 1,
            "shadowRadius": 0,
            "width": 40,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[` 3`] = `
<View
  style={
    Object {
      "backgroundColor": "#00b4d5",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "height": "50%",
            "justifyContent": "center",
          },
          Object {
            "height": "35%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
          }
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.1)",
              "borderRadius": 9999999,
              "margin": 20,
            }
          }
        >
          <Image
            collapsable={undefined}
            resizeMode="cover"
            source={1337}
            style={
              Object {
                "borderRadius": 100,
                "height": 200,
                "margin": 20,
                "width": 200,
              }
            }
          />
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginLeft": -70,
              "marginRight": 20,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#73dbff",
                "borderRadius": 9999,
                "elevation": 10,
                "height": 55,
                "justifyContent": "center",
                "padding": 5,
                "shadowColor": "rgba(0, 0, 0, 0.1)",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 4,
                },
                "shadowOpacity": 1,
                "shadowRadius": 0,
                "width": 55,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": "50%",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderColor": "#ffffff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Object {
                  "borderRadius": 7,
                  "margin": 5,
                  "padding": 5,
                },
              ],
              Object {},
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": NaN,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": NaN,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              Apfel
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderColor": "#ffffff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Object {
                  "borderRadius": 7,
                  "margin": 5,
                  "padding": 5,
                },
              ],
              Object {},
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": NaN,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": NaN,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#ffffff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              Nase
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.2)",
                "borderColor": "#73dbff",
                "borderRadius": 9999,
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#73dbff",
                  "elevation": 5,
                  "justifyContent": "center",
                  "opacity": 1,
                  "shadowColor": "rgba(0, 0, 0, 0.1)",
                  "shadowOffset": Object {
                    "height": 4,
                    "width": 4,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 0,
                },
                Object {
                  "borderRadius": 7,
                  "margin": 5,
                  "padding": 5,
                },
              ],
              Object {
                "backgroundColor": "#ffffff",
              },
            ]
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "alignItems": "center",
                "height": NaN,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": NaN,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#73dbff",
                  "fontFamily": "norddruck",
                  "fontSize": 35,
                  "marginTop": 5,
                  "textAlign": "center",
                }
              }
            >
              Ball
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "top": 35,
        },
        Object {
          "right": 15,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#73dbff",
            "borderRadius": 9999,
            "elevation": 10,
            "height": 40,
            "justifyContent": "center",
            "padding": 5,
            "shadowColor": "rgba(0, 0, 0, 0.1)",
            "shadowOffset": Object {
              "height": 4,
              "width": 4,
            },
            "shadowOpacity": 1,
            "shadowRadius": 0,
            "width": 40,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
</View>
`;
